### PR TITLE
Deprecates inferno state hook

### PR DESCRIPTION
**Pull request by jlsnow301**
## About The Pull Request
Marks useLocalState as deprecated. Removes a lot of its usages. I even wrote some documentation to boot!

This is no longer necessary, as it was Inferno specific. Now that we have real hooks, use them!
```tsx
// Old
const [someState, setSomeState] = useLocalState(context, 'some_state_identifier', false);

// Now
const [someState, setSomeState] = useState(false);
```

Read more! Or the rest of the React documentation, it's great: [React State](https://react.dev/learn/state-a-components-memory)

## Why It's Good For The Game
We don't need uselocalstate anymore. Easier to read and probably faster this way

## Changelog

N/A nothing player facing

### CODE REVIEW

File: tgui/packages/tgui/interfaces/TramControl.jsx [hunk 1]
```diff
@@ -39,91 +40,12 @@ const dipUnderCircle = (dest, dep) => {
...
-                    }}
-                  />
-                </Stack.Item>
-              ))}
-            </Stack>
-          </Stack.Item>
-        )}
-      </Stack>
-    );
-  };
```
Comment (jlsnow301) [inline tgui/packages/tgui/interfaces/TramControl.jsx:126]: do NOT nest components like this

**Comment by MrStonedOne** [comment]
> Shared state UI is a bad design. There's nothing good about an invisible user clicking your screen.

Hard disagree. the whole point of this system was to match how it worked before tgui, which was that the UI for computers and machinery was the rendering of a specific display that existed in world, and ofcourse if somebody comes along they can see what you are doing and what you have typed and also fuck with your shit by mashing on the keyboard or clicking the mouse. 

This was intentional and should stick around.

**Comment by jlsnow301** [comment]
> **This was intentional and should stick around.** 

I am under the impression most of these UIs used it unintentionally. Changeling antag info. 

If there are any in specific you think are intentional and you want them to remain annoying, I can leave the shared state for them

**Comment by Mothblocks** [comment]
just leave the shared state for now, i agree that it's a pretty cool way of seeing it. but we should have a better system for restricting what gets sent through them that is required to be setup, while also locally predicting the state.

**Comment by MrStonedOne** [comment]
they are correct that ooc interfaces shouldn't be using shared state like that, just ic interfaces. i didn't look at the pr deeply to understand what is or isn't getting it removed.

**Pull request merged**